tests/backup/backup_test.go

```diff
@@ -61,8 +84,296 @@ func TestBackup(t *testing.T) {
+			// TODO(stgleb): Add multi-namespace backup when ready in px-backup
+			for _, namespace := range bkpNamespaces {
+				backupName := fmt.Sprintf("%s-%s", BackupName, namespace)
+				logrus.Infof("Create backup full name %s:%s:%s",
+					SourceClusterName, namespace, backupName)
+				CreateBackup(backupName,
+					SourceClusterName, BLocationName,
+					[]string{namespace}, labelSelectores, orgID)
+			}
+		})
+
+		Step("Kill stork", func() {
+			// setup task to delete stork pods as soon as it starts doing backup
+			eventCheck := func() (bool, error) {
+				for _, namespace := range bkpNamespaces {
+					backupName := fmt.Sprintf("%s-%s", BackupName, namespace)
+					req := &api.BackupInspectRequest{
+						Name:  backupName,
+						OrgId: orgID,
+					}
+
+					logrus.Infof("backup %s wait for running", backupName)
+					err := Inst().Backup.WaitForRunning(context.Background(),
+						req, time.Millisecond, time.Millisecond)
+
+					if err != nil {
+						logrus.Infof("backup %s wait for running err %v",
+							backupName, err)
+
+						continue
+					} else {
+						logrus.Infof("backup %s is running", backupName)
+						return true, nil
+					}
+				}
+				return false, nil
+			}
+
+			deleteOpts := &scheduler.DeleteTasksOptions{
+				TriggerOptions: driver_api.TriggerOptions{
+					TriggerCb:            eventCheck,
+					TriggerCheckInterval: triggerCheckInterval,
+					TriggerCheckTimeout:  triggerCheckTimeout,
+				},
+			}
+
+			t := func(interval sched.Interval) {
+				ctx := &scheduler.Context{
+					App: &spec.AppSpec{
+						SpecList: []interface{}{
+							&appsapi.Deployment{
+								ObjectMeta: meta_v1.ObjectMeta{
+									Name:      storkDeploymentName,
+									Namespace: storkDeploymentNamespace,
+								},
+							},
+						},
+					},
+				}
+
+				err := Inst().S.DeleteTasks(ctx, deleteOpts)
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			taskID, err := sched.Instance().Schedule(t,
+				sched.Periodic(time.Second),
+				time.Now(), true)
+			Expect(err).NotTo(HaveOccurred())
+			defer sched.Instance().Cancel(taskID)
+		})
+
+		Step(fmt.Sprintf("Wait for Backup [%s] to complete", BackupName), func() {
+			for _, namespace := range bkpNamespaces {
+				backupName := fmt.Sprintf("%s-%s", BackupName, namespace)
+				logrus.Infof("Wait for backup %s to complete", backupName)
+				err := Inst().Backup.WaitForBackupCompletion(
+					context.Background(),
+					backupName, orgID,
+					BackupRestoreCompletionTimeoutMin*time.Minute,
+					RetrySeconds*time.Second)
+				Expect(err).NotTo(HaveOccurred(),
+					fmt.Sprintf("Failed to wait for backup [%s] to complete. Error: [%v]",
+						BackupName, err))
+			}
+		})
+
+		Step(fmt.Sprintf("Create Restore [%s]", RestoreName), func() {
+			for _, namespace := range bkpNamespaces {
+				backupName := fmt.Sprintf("%s-%s", BackupName, namespace)
+				restoreName := fmt.Sprintf("%s-%s", RestoreName, namespace)
+				logrus.Infof("Create restore %s:%s:%s from backup %s:%s:%s",
```

Comment:
We can move this description to Step() description itself so that we can remove the duplicate log here

Reply:
Well, that won't work that way. I need to print every namespace and every restore created this needs to be loged from the loop.

Reply:
Got it